visualization/app/codeCharta/ui/mapTreeView/mapTreeView.component.ts

```diff
@@ -1,27 +1,86 @@
 import { IRootScopeService, ITimeoutService } from "angular"
-import { CodeMapNode } from "../../codeCharta.model"
+import { CodeMapNode, NodeType, SortingOption } from "../../codeCharta.model"
 import { CodeMapPreRenderService, CodeMapPreRenderServiceSubscriber } from "../codeMap/codeMap.preRender.service"
+import { StoreService } from "../../state/store.service"
+import {
+	SortingOrderAscendingService,
+	SortingOrderAscendingSubscriber
+} from "../../state/store/appSettings/sortingOrderAscending/sortingOrderAscending.service"
+import { setSortingOrderAscending } from "../../state/store/appSettings/sortingOrderAscending/sortingOrderAscending.actions"
+import { SortingOptionService, SortingOptionSubscriber } from "../../state/store/dynamicSettings/sortingOption/sortingOption.service"
+import { setSortingOption } from "../../state/store/dynamicSettings/sortingOption/sortingOption.actions"
 
-export class MapTreeViewController implements CodeMapPreRenderServiceSubscriber {
+export class MapTreeViewController implements CodeMapPreRenderServiceSubscriber, SortingOptionSubscriber, SortingOrderAscendingSubscriber {
 	private _viewModel: {
 		rootNode: CodeMapNode
 	} = {
 		rootNode: null
 	}
 
 	/* @ngInject */
-	constructor(private $rootScope: IRootScopeService, private $timeout: ITimeoutService) {
+	constructor(private $rootScope: IRootScopeService, private $timeout: ITimeoutService, private storeService: StoreService) {
 		CodeMapPreRenderService.subscribe(this.$rootScope, this)
+		SortingOptionService.subscribe(this.$rootScope, this)
+		SortingOrderAscendingService.subscribe(this.$rootScope, this)
+	}
+
+	public onSortingOptionChanged(sortingOption: SortingOption) {
```

Comment:
```
public onSortingOptionChanged(sortingOption: SortingOption) {
		if (sortingOption === SortingOption.NUMBER_OF_FILES) {
			this._viewModel.rootNode = this.applySortOrderChange(
				this._viewModel.rootNode,
				(a, b) => b.attributes["unary"] - a.attributes["unary"],
				false
			)
		} else {
this._viewModel.rootNode = this.applySortOrderChange(this._viewModel.rootNode, (a, b) => (b.name > a.name ? 0 : 1), false)
                 }	this.storeService.dispatch(setSortingOrderAscending(this.storeService.getState().appSettings.sortingOrderAscending))
	}
```

what about this solution?

Reply:
will desync the button if you load a map without changes
I will commit another fix now, that should resolve the button issue